src/server/game/Handlers/ChatHandler.cpp

```diff
@@ -568,20 +569,19 @@ void WorldSession::HandleMessagechatOpcode(WorldPacket& recvData)
     }
 }
 
-void WorldSession::HandleEmoteOpcode(WorldPacket& recvData)
+void WorldSession::HandleEmoteOpcode(WorldPackets::Chat::EmoteClient& packet)
 {
-    if (!GetPlayer()->IsAlive() || GetPlayer()->HasUnitState(UNIT_STATE_DIED))
```

Comment:
it's easier to review a PR if there are very few changes. this one looks more like some refactoring of the whole function than just using the new packt structure.

Reply:
Actually this is fine. The check was previously blocking the packet reading but now that the packet is always being read the check was just looking ugly up here so he moved it down to the other check. Just a cosmetic improvement, not a refactor.